web/source/osk/visualKeyboard.ts

```diff
@@ -1491,19 +1523,48 @@ namespace com.keyman.osk {
       kbd.style.border = '1px solid #ccc';
 
       // Once the element is inserted into the DOM, refresh the layout so that proper text scaling may apply.
-      const refreshInterval = window.setInterval(function () {
-        let computedStyle = getComputedStyle(kbd);
-        if (computedStyle.fontSize) {
-          if (kbd.style.fontSize) {
-            // Preserve the new setting (provided by CSS)
-            kbdObj.fontSize = new ParsedLengthStyle(kbd.style.fontSize);
+      const detectAndHandleInsertion = () => {
+        if(document.contains(kbd)) {
+          // Yay, insertion!
+
+          try {
+            // Are there font-size attributes we may safely adjust?  If so, do that!
+            if(getComputedStyle(kbd) && kbd.style.fontSize) {
+              kbdObj.fontSize = new ParsedLengthStyle(kbd.style.fontSize);
+            }
+
+            // Make sure that the stylesheet is attached, now that the keyboard-doc's been inserted.
+            // The stylesheet is currently built + constructed in the same code that attaches it to
+            // the page.
+            kbdObj.appendStyleSheet();
+
+            // Grab a reference to the stylesheet.
+            const stylesheet = kbdObj.styleSheet;
+            const stylesheetHead = stylesheet.parentElement;
+
+            // Don't reset top-level stuff; just the visible layer.
+            kbdObj.currentLayer.refreshLayout(kbdObj, kbdObj.height);
+
+            // We no longer need a reference to the constructing VisualKeyboard, so we should let
+            // it clean up its <head> stylesheet links.  This detaches the stylesheet, though.
+            kbdObj.shutdown();
```

Comment:
Is this an opportunity for future cleanup? Why do we need to do this little dance of attach-detach-re-attach?

Reply:
If we didn't `shutdown`, we wouldn't need to re-attach.  Actually, in the current form of the codebase, I don't know that it does anything extra.

Still, since there are times where we do need to have aspects manually turned off when discarding an instance, I thought consistency in instance disposal would be best.